FirebaseStorage/Tests/SwiftIntegration/StorageIntegration.swift

```diff
@@ -0,0 +1,672 @@
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("hello.txt")
+    do {
+      try data.write(to: fileURL, options: Data.WritingOptions.atomicWrite)
+    } catch {
+      XCTFail("Write to \(fileURL) failed")
+    }
+    ref?.putFile(from: fileURL, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      XCTAssertEqual(fileName, metadata?.name)
+      ref?.getMetadata(completion: { metadata, error in
+        XCTAssertNotNil(metadata, "Metadata should not be nil")
+        XCTAssertNil(error, "Error should be nil")
+        XCTAssertEqual(fileName, metadata?.name)
+        expectation.fulfill()
+      })
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutDataNoMetadata() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutDataNoMetadata")
+
+    let ref = storage?.reference(withPath: "ios/public/testUnauthenticatedSimplePutDataNoMetadata")
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    ref?.putData(data, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutFileNoMetadata() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutFileNoMetadata")
+
+    let fileName = "hello&+@_ .txt"
+    let ref = storage?.reference(withPath: "ios/public/" + fileName)
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("hello.txt")
+    do {
+      try data.write(to: fileURL, options: Data.WritingOptions.atomicWrite)
+    } catch {
+      XCTFail("Write to \(fileURL) failed")
+    }
+    ref?.putFile(from: fileURL, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimpleGetData() {
+    let expectation = self.expectation(description: "testUnauthenticatedSimpleGetData")
+
+    let ref = storage?.reference(withPath: "ios/public/1mb")
+    ref?.getData(maxSize: 1024 * 1024, completion: { data, error in
+      XCTAssertNotNil(data, "Data should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimpleGetDataInBackgroundQueue() {
+    let expectation = self.expectation(description: "testUnauthenticatedSimpleGetDataInBackgroundQueue")
```

Comment:
Nit that you don't have to change: For here and other places, you can use `#function` to get the full function name (it'll include extras, including the class I think?) but for this purpose it'd probably be fine.

Reply:
Done.